### PR TITLE
Issue #119: Ignore messages not related to request

### PR DESCRIPTION
**Pull request by ihorserba**

### CODE REVIEW

File: examples/js/index.js [hunk 1]
```diff
@@ -54,13 +59,23 @@ function zelloStartStream(ws, opusStream, onCompleteCb) {
         "packet_duration": opusStream.packetDurationMs,
     }));
 
+    const startTimeoutMs = 2000;
+    var startTimeout = setTimeout(onCompleteCb, startTimeoutMs, null);
```
Comment (vocoded) [inline examples/js/index.js:63]: use `let`

Comment (vocoded) [reply examples/js/index.js:63]: will it be possible to have multiple timeouts running?

Comment (vocoded) [reply examples/js/index.js:63]: does the oncomplete callback remove the listener from the websocket onmessage event?

File: examples/py/main.py [hunk 2]
```diff
@@ -67,16 +68,17 @@ async def zello_stream_audio_to_channel(username, password, token, channel, opus
...
                 ZelloStreamID = stream_id
                 print(f"Started streaming {opusfile}")
                 await zello_stream_send_audio(session, ws, stream_id, opus_stream)
-                await zello_stream_stop(ws, stream_id)
+                await asyncio.wait_for(zello_stream_stop(ws, stream_id), WS_TIMEOUT_SEC)
     except (NameError, aiohttp.client_exceptions.ClientError, IOError) as error:
             print(error)
-
+    except asyncio.TimeoutError:
+        print("Communication timeout")
```
Comment (vocoded) [inline examples/py/main.py:81]: additional cleanup needed?

File: examples/js/index.js (hunk 1 above)
Comment (ihorserba) [reply examples/js/index.js]: > use `let`

Replaced `var` with `let` over the file.



> will it be possible to have multiple timeouts running?

All timeouts are being cleared before onCompleteCb() is called.
And onCompleteCb() is the only way to go further.

> does the oncomplete callback remove the listener from the websocket onmessage event?

That's a good point - it didn't remove the listener, I have fixed it.

File: examples/py/main.py (hunk 2 above)
Comment (ihorserba) [reply examples/py/main.py:81]: I don't think so. 
The only cleanup function here is `zello_stream_stop()`, but we have to call it after we've got stream_id from the server.
Possible timed out functions here are :
`authenticate()` - no stream_id yet
`zello_stream_start()` - returns stream_id, so if timed out - there is no stream_id, 
`zello_stream_stop()` - no reason to repeat the same call

File: examples/cs/ZelloMediaStream.cs [hunk 3]
```diff
@@ -93,18 +95,40 @@ private bool SendJson(dynamic json)
...
                 }
+                catch
+                {
+                    // Received not a JSON - keep listening
+                    continue;
+                }
+            }
+        }
+
+        private dynamic ReceiveJson()
```
Comment (vocoded) [inline examples/cs/ZelloMediaStream.cs:118]: this should return a ResponseJson, correct?

File: examples/js/index.js [hunk 4]
```diff
@@ -18,14 +18,19 @@ function zelloAuthorize(ws, opusStream, username, password, token, channel, onCo
...
+    let isAuthorized = false, isChannelAvailable = false;
+    let authTimeout = setTimeout(onCompleteCb, authTimeoutMs, false);
     ws.onmessage = function(event) {
-        let json = JSON.parse(event.data);
-        if (json.refresh_token) {
-            isAuthorized = true;
-        } else if (json.command === "on_channel_status" && json.status === "online") {
-            isChannelAvailable = true;
+        try {
+            let json = JSON.parse(event.data);
```
Comment (vocoded) [inline examples/js/index.js:25]: this can be const

File: examples/js/index.js [hunk 5]
```diff
@@ -35,15 +40,15 @@ function zelloAuthorize(ws, opusStream, username, password, token, channel, onCo
...
+    let codecHeaderRaw = new Uint8Array(4);
     codecHeaderRaw[2] = opusStream.framesPerPacket;
     codecHeaderRaw[3] = opusStream.packetDurationMs;
 
     // sampleRate is represented in two bytes in little endian.
     // https://github.com/zelloptt/zello-channel-api/blob/409378acd06257bcd07e3f89e4fbc885a0cc6663/sdks/js/src/classes/utils.js#L63
     codecHeaderRaw[0] = parseInt(opusStream.sampleRate & 0xff, 10);
     codecHeaderRaw[1] = parseInt(opusStream.sampleRate / 0x100, 10) & 0xff;
-    var codecHeader = Buffer.from(codecHeaderRaw).toString('base64');
+    let codecHeader = Buffer.from(codecHeaderRaw).toString('base64');
```
Comment (vocoded) [inline examples/js/index.js:51]: these can both be const

File: examples/js/index.js [hunk 6]
```diff
@@ -54,13 +59,23 @@ function zelloStartStream(ws, opusStream, onCompleteCb) {
...
     }));
 
+    const startTimeoutMs = 2000;
+    let startTimeout = setTimeout(onCompleteCb, startTimeoutMs, null);
     ws.onmessage = function(event) {
-        let json = JSON.parse(event.data);
-        if (json && json.success && json.stream_id) {
-            return onCompleteCb(json.stream_id);
+        try {
+            let json = JSON.parse(event.data);
```
Comment (vocoded) [inline examples/js/index.js:66]: can be const

File: examples/js/index.js [hunk 7]
```diff
@@ -134,7 +149,13 @@ function zelloStopStream(ws, streamId) {
 }
 
 function zelloStreamReadyCb(opusStream, username, password, token, channel) {
-    var ws = new WebSocket("wss://zello.io/ws");
+    let ws = new WebSocket("wss://zello.io/ws");
```
Comment (vocoded) [inline examples/js/index.js:152]: can be const

File: examples/py/main.py (hunk 2 above)
Comment (vocoded) [reply examples/py/main.py:81]: do we need to close the websocket in auth times out?  if we time out a stream operation does the application simply continue?

Comment (ihorserba) [reply examples/py/main.py:81]: > do we need to close the websocket in auth times out? if we time out a stream operation does the application simply continue?

An exception raised inside the `with` blocks:

```
        async with aiohttp.ClientSession(connector = conn) as session:
            async with session.ws_connect(WS_ENDPOINT) as ws:
```
would cleanup `ws` and `session`.
The `with` statement is a syntactic sugar for try-except-finally logic:
https://www.python.org/dev/peps/pep-0343/

Finally I tested it using `WS_TIMEOUT_SEC = 0.1` and got "Communication timeout" message and the application stopped.

File: examples/cs/ZelloMediaStream.cs (hunk 3 above)
Comment (ihorserba) [reply examples/cs/ZelloMediaStream.cs:118]: Right, ResponseJson or null if timed out

File: examples/js/index.js (hunk 4 above)
Comment (ihorserba) [reply examples/js/index.js]: changed to const where possible over a file

File: examples/js/index.js (hunk 5 above)
Comment (ihorserba) [reply examples/js/index.js]: changed to const where possible over a file

File: examples/js/index.js (hunk 6 above)
Comment (ihorserba) [reply examples/js/index.js]: changed to const where possible over a file

File: examples/js/index.js (hunk 7 above)
Comment (ihorserba) [reply examples/js/index.js]: changed to const where possible over a file

File: examples/cs/ZelloMediaStream.cs (hunk 3 above)
Comment (vocoded) [reply examples/cs/ZelloMediaStream.cs:118]: so can we change to the more specific type?

File: examples/py/main.py (hunk 2 above)
Comment (vocoded) [reply examples/py/main.py:81]: thanks

File: examples/cs/ZelloMediaStream.cs (hunk 3 above)
Comment (ihorserba) [reply examples/cs/ZelloMediaStream.cs]: changed

**Pull request merged**